### ISSUE TITLE
scipy1.8.0 breaks sparse matrix multiplication

### ISSUE THREAD

**Issue by auggiemarignier** [auggiemarignier/pxmcmc#6]
Get this error on multiple tests with `scipy==1.8.0` but not `scipy==1.7.3`

```
self = <190x190 sparse matrix of type '<class 'numpy.float64'>'
        with 190 stored elements (1 diagonals) in DIAgonal format>
other = 99.99999999999999

    def __matmul__(self, other):
        if isscalarlike(other):
>           raise ValueError("Scalar operands are not allowed, "
                             "use '*' instead")
E           ValueError: Scalar operands are not allowed, use '*' instead
```

Quick fix is to restrict `scipy` dependency to `1.7.3`.  Longer term, figure out what this means.

**Comment by auggiemarignier** [comment]
Quick fix done `6e01dede9c925c54c57bd491bcd4b733ba0dd9d3`

**Comment by auggiemarignier** [comment]
fixed as of `scipy 1.8.1` (https://github.com/scipy/scipy/pull/16035).  Making minimum requirement `v1.9.2` in `pyproject.toml`  
`e4d2dce6a26876d6367f339072295ad170ea09c4`
